homeassistant/components/group/entity.py

```diff
@@ -430,12 +459,14 @@ def _async_update_group_state(self, tr_state: State | None = None) -> None:
         # have the same on state we use this state
         # and its hass.data[REG_KEY].on_off_mapping to off
         if num_on_states == 1:
-            on_state = list(self._on_states)[0]
+            on_state = next(iter(self._on_states))
         # If we do not have an on state for any domains
         # we use None (which will be STATE_UNKNOWN)
         elif num_on_states == 0:
             self._state = None
             return
+        if self.single_state_type_key:
+            on_state = self.single_state_type_key[0]
```

Comment:
Maybe we should make `single_state_type_key` a slotted frozen `dataclass` so anyone who looks at this code in a the future can figure out this is getting the on state?

Reply:
done